### PR TITLE
Add interface in ConfigIService

### PR DESCRIPTION
**Pull request by liuminghui233**
## Description

Add some interfaces in ConfigIService, which are used in IPartitionFetcher.

### CODE REVIEW

File: thrift-confignode/src/main/thrift/confignode.thrift [hunk 1]
```diff
@@ -21,6 +21,11 @@ include "rpc.thrift"
 namespace java org.apache.iotdb.confignode.rpc.thrift
 namespace py iotdb.thrift.confignode
 
+struct EndPoint {
```
Comment (CRZbulabula) [inline thrift-confignode/src/main/thrift/confignode.thrift:24]: The EndPoint is already defined in rpc.thrift. And I've included it before. Why not just use it?

**Comment by CRZbulabula** [comment]
I noticed that there are more interfaces defined in IPartitionFetcher. Why not add them to the confignode.thrift in this PR?

By the way, you should also implement the empty interfaces in ConfigNodeRPCServerProcessor. Otherwise, the code couldn't compile successfully. Finally, I will fill these interfaces later.

File: thrift-confignode/src/main/thrift/confignode.thrift (hunk 1 above)
Comment (liuminghui233) [reply thrift-confignode/src/main/thrift/confignode.thrift:24]: fixed ~

**Comment by liuminghui233** [comment]
@CRZbulabula @MarcosZyk @MrQuansy @JackieTien97 PTAL :)

File: thrift-confignode/src/main/thrift/confignode.thrift [hunk 2]
```diff
@@ -76,6 +76,40 @@ struct DeviceGroupHashInfo {
...
+
+struct DataPartitionInfoResp {
+    // Map<StorageGroup, Map<DeviceGroupID, Map<TimePartitionId, List<DataRegionReplicaInfo>>>>
+    1: required map<string, map<i32, map<i64, list<RegionInfo>>>> dataPartitionMap
+}
+
+struct SchemaPartitionInfoResp {
+    // Map<StorageGroup, Map<DeviceGroupID, SchemaRegionPlaceInfo>>
+    1: required map<string, map<i32, RegionInfo>> schemaPartitionInfo
+}
```
Comment (MarcosZyk) [inline thrift-confignode/src/main/thrift/confignode.thrift:104]: The deviceId of i32 can't support wildcard match of ParthPattern. Maybe should replace it with the raw devicePath

Comment (MarcosZyk) [inline thrift-confignode/src/main/thrift/confignode.thrift:104]: The deviceId of i32 can't support wildcard match of ParthPattern. Maybe should replace it with the raw devicePath

Comment (JackieTien97) [reply thrift-confignode/src/main/thrift/confignode.thrift]: If you have `*` in devicePath, you will need to send your request to all the RegionInfo.

Comment (liuminghui233) [reply thrift-confignode/src/main/thrift/confignode.thrift]: same as above

**Comment by MrQuansy** [comment]
I noticed that the new interface in ConfigIService, the parameter of fetch data partition is device paths instead of device group ids. At first, we hope data node can execute the hash calculation to decrease calculation workload on config nodes.
So I think we can replace the device path with device group ids, or offering another interface.

**Review by JackieTien97 — APPROVED** [review]

**Comment by JackieTien97** [comment]
> I noticed the new interface in ConfigIService. The parameter of fetching data partition and fetching schema partition are the device paths instead of device group ids. At first, we hope data node can execute the hash calculation to decrease calculation workload on config nodes.
> 
> So I think we can replace the device path with device group ids, or offering another interface. We can discuss afterwards.

When we have `*` in our path, we can't calculate the DeviceGroupId.

**Comment by MarcosZyk** [comment]
> > I noticed the new interface in ConfigIService. The parameter of fetching data partition and fetching schema partition are the device paths instead of device group ids. At first, we hope data node can execute the hash calculation to decrease calculation workload on config nodes.
> > So I think we can replace the device path with device group ids, or offering another interface. We can discuss afterwards.
> 
> When we have `*` in our path, we can't calculate the DeviceGroupId.

Got it. i32 will be ok.

File: confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java [hunk 3]
```diff
@@ -156,6 +162,22 @@ public DataPartitionInfo getDataPartition(GetDataPartitionReq req) throws TExcep
...
 
+  @Override
+  public DataPartitionInfoResp fetchDataPartitionInfo(FetchDataPartitionReq req) throws TException {
+    return null;
+  }
+
+  @Override
+  public SchemaPartitionInfoResp fetchSchemaPartitionInfo(FetchSchemaPartitionReq req)
+      throws TException {
+    return null;
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java:173]: Will return null cause trouble? Might it be better to return a resp with empty maps?

File: confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java [hunk 4]
```diff
@@ -156,6 +162,22 @@ public DataPartitionInfo getDataPartition(GetDataPartitionReq req) throws TExcep
...
+  }
+
+  @Override
+  public PartitionInfoResp fetchPartitionInfo(FetchPartitionReq req) throws TException {
+    return null;
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java:178]: Will return null cause trouble? Might it be better to return a resp with empty maps?

**Review by CRZbulabula — APPROVED** [review]
LGTM.

File: confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java (hunk 4 above)
Comment (liuminghui233) [reply confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/server/ConfigNodeRPCServerProcessor.java:178]: These methods have not been used yet. It's nothing serious ~

**Pull request merged**